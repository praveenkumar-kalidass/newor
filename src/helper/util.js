import CONSTANT from 'constant';

export const formatCurrency = (value) => {
  const formatter = new Intl.NumberFormat('hi-IN', { style: 'currency', currency: 'INR' });
  return formatter.format(value);
};

export const unformatCurrency = (value) => {
  const withoutSymbol = value.replace(CONSTANT.APP_LITERAL.RUPEE_SYMBOL, '');
  const currency = withoutSymbol.replace(/,/g, '');
  return parseFloat(currency);
};

export const formatUrlParams = (data) => Object.keys(data).map((key) => (
  `${encodeURIComponent(key)}=${encodeURIComponent(data[key])}`
)).join('&');
