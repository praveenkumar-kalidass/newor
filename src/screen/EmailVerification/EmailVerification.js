import React, { useCallback, useEffect, useState } from 'react';
import { useRoute, useNavigation } from '@react-navigation/native';
import {
  Flex, Center, Image, Text,
} from 'native-base';

import CONSTANT from 'constant';
import ROUTE from 'constant/route';
import useTheme from 'theme/useTheme';
import useTranslation from 'translation/useTranslation';
import useUser from 'api/useUser';
import withBackground from 'helper/withBackground';
import { Background1 } from 'component/Background';
import TitleBar from 'component/TitleBar/TitleBar';
import AppButton from 'component/AppButton';
import Translation from 'translation/Translation';
import NEWOR_SUCCESS from 'asset/image/newor-success.png';
import NEWOR_FAILURE from 'asset/image/newor-failure.png';
import { Container, Title } from './EmailVerification.style';

const EmailVerification = () => {
  const [isSuccess, setIsSuccess] = useState(false);
  const [isError, setIsError] = useState(false);
  const { params } = useRoute();
  const { color } = useTheme();
  const navigation = useNavigation();
  const { translate } = useTranslation();
  const { verify } = useUser();

  const doVerifyUser = useCallback(async () => {
    try {
      await verify({ token: params?.token });
      setIsSuccess(true);
    } catch (error) {
      setIsError(true);
    }
  }, []);

  useEffect(() => {
    doVerifyUser();
  }, []);

  return (
    <Flex flex={1}>
      <Flex flex={1} direction="row" p="5" justify="center">
        <If condition={isSuccess || isError}>
          <Container testID="email-verification">
            <TitleBar color={color.SECONDARY_100} styleProps={{ mb: 5 }}>
              <Title>
                {isSuccess && translate('SUCCESS')}
                {isError && translate('ERROR')}
                !
              </Title>
            </TitleBar>
            <Center>
              <If condition={isSuccess}>
                <Image alt={CONSTANT.APP_NAME} source={NEWOR_SUCCESS} size={250} />
              </If>
              <If condition={isError}>
                <Image alt={CONSTANT.APP_NAME} source={NEWOR_FAILURE} size={250} />
              </If>
            </Center>
            <If condition={isSuccess}>
              <Translation tkey="EMAIL_VERIFICATION_SUCCESS" as={Text} mb={5} />
              <Translation
                tkey="LOGIN"
                as={AppButton}
                variant="primary"
                mb={5}
                testID="login-submit"
                onPress={() => navigation.navigate(ROUTE.LOGIN)}
              />
            </If>
            <If condition={isError}>
              <Translation tkey="EMAIL_VERIFICATION_FAILURE" as={Text} mb={5} />
            </If>
          </Container>
        </If>
      </Flex>
    </Flex>
  );
};

export default withBackground({ Component: EmailVerification, Background: Background1 });
