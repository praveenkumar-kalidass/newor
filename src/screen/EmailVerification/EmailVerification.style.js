import styled from 'styled-components/native';
import { Box, Text } from 'native-base';

export const Container = styled(Box).attrs(({ theme }) => ({
  borderRadius: 20,
  flex: 1,
  alignSelf: 'center',
  p: 5,
  bg: theme.color.MODAL,
  elevation: 1,
  shadowColor: theme.color.BACKGROUND_0,
}))``;

export const Title = styled(Text).attrs(() => ({
  bold: true,
  fontSize: 20,
}))``;
