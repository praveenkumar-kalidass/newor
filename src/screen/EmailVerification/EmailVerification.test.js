import React from 'react';

import { render, waitFor } from 'test/util';
import EmailVerification from './EmailVerification';

jest.mock('@react-navigation/native', () => {
  const reactNavigation = jest.requireActual('@react-navigation/native');
  return {
    ...reactNavigation,
    useRoute: () => ({ params: 'testtoken123' }),
  };
});
const mockUseUser = {
  verify: jest.fn(),
};
jest.mock('api/useUser', () => () => mockUseUser);

describe('EmailVerification', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match success snapshot', async () => {
    mockUseUser.verify.mockResolvedValueOnce();

    const container = await render(<EmailVerification />);
    const { queryByTestId } = container;
    await waitFor(() => queryByTestId('email-verification'));

    expect(container).toMatchSnapshot();
  });

  it('should match error snapshot', async () => {
    mockUseUser.verify.mockRejectedValueOnce();

    const container = render(<EmailVerification />);
    const { queryByTestId } = container;
    await waitFor(() => queryByTestId('email-verification'));

    expect(container).toMatchSnapshot();
  });
});
