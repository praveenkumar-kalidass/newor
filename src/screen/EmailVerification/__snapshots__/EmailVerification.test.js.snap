// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailVerification should match error snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Array [
          Object {
            "borderWidth": 0,
            "left": 0,
            "position": "absolute",
            "top": 0,
          },
          Object {
            "height": 1334,
            "width": 750,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "position": "absolute",
            },
          ]
        }
      >
        <RNSVGSvgView
          bbHeight={1334}
          bbWidth={750}
          focusable={false}
          height={1334}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "flex": 0,
                "height": 1334,
                "width": 750,
              },
            ]
          }
          width={750}
        >
          <RNSVGGroup>
            <RNSVGPath
              d="M0 200 L0 300 L750 200 L750 100 L0 200 Z"
              fill={4289527808}
              propList={
                Array [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M0 350 L0 370 L750 270 L750 250 L0 350 Z"
              fill={4289527808}
              propList={
                Array [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M0 1084 L0 1104 L750 1004 L750 984 L0 1084 Z"
              fill={4278199178}
              propList={
                Array [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M0 1154 L0 1254 L750 1154 L750 1054 L0 1154 Z"
              fill={4278199178}
              propList={
                Array [
                  "fill",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
      <View
        dataSet={Object {}}
        style={
          Object {
            "display": "flex",
            "flex": 1,
            "flexDirection": "column",
          }
        }
      >
        <View
          dataSet={Object {}}
          style={
            Object {
              "display": "flex",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingBottom": 20,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 20,
            }
          }
        >
          <View
            dataSet={Object {}}
            elevation={1}
            shadowColor="#000000"
            shadowOffset={
              Object {
                "height": 1,
                "width": 0,
              }
            }
            shadowOpacity={0.8}
            shadowRadius={1}
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "#eeeeee",
                  "borderRadius": 20,
                  "flex": 1,
                  "paddingBottom": 20,
                  "paddingLeft": 20,
                  "paddingRight": 20,
                  "paddingTop": 20,
                },
                Array [
                  Object {},
                ],
              ]
            }
            testID="email-verification"
          >
            <View
              dataSet={Object {}}
              style={
                Array [
                  Object {
                    "display": "flex",
                    "flexDirection": "row",
                    "marginBottom": 20,
                  },
                  Array [
                    Object {},
                  ],
                ]
              }
              styleProps={
                Object {
                  "mb": 5,
                }
              }
            >
              <View
                dataSet={Object {}}
                style={
                  Array [
                    Object {
                      "backgroundColor": "#00238a",
                      "color": "#00238a",
                      "marginRight": 8,
                      "paddingRight": 8,
                    },
                    Array [
                      Object {},
                    ],
                  ]
                }
              />
              <Text
                dataSet={Object {}}
                style={
                  Array [
                    Object {
                      "backgroundColor": undefined,
                      "color": "#171717",
                      "fontFamily": undefined,
                      "fontSize": 20,
                      "fontStyle": "normal",
                      "fontWeight": "700",
                      "letterSpacing": 0,
                      "lineHeight": 30,
                      "textDecorationLine": undefined,
                    },
                    Array [
                      Object {},
                    ],
                  ]
                }
              >
                ERROR
                !
              </Text>
            </View>
            <View
              dataSet={Object {}}
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "justifyContent": "center",
                }
              }
            >
              <Image
                accessibilityLabel="newor"
                alt="newor"
                dataSet={Object {}}
                onError={[Function]}
                source={
                  Object {
                    "testUri": "../../../src/asset/image/newor-failure.png",
                  }
                }
                style={
                  Object {
                    "height": 250,
                    "maxWidth": "100%",
                    "width": 250,
                  }
                }
              />
            </View>
            <Text
              dataSet={Object {}}
              style={
                Object {
                  "backgroundColor": undefined,
                  "color": "#171717",
                  "fontFamily": undefined,
                  "fontSize": 14,
                  "fontStyle": "normal",
                  "fontWeight": "400",
                  "letterSpacing": 0,
                  "lineHeight": 21,
                  "marginBottom": 20,
                  "textDecorationLine": undefined,
                }
              }
            >
              EMAIL_VERIFICATION_FAILURE
            </Text>
          </View>
        </View>
      </View>
      <View
        pointerEvents="none"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            undefined,
          ]
        }
      />
    </View>
    <View
      elevation={0}
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Array [
          Object {
            "borderWidth": 0,
            "left": -600,
            "position": "absolute",
            "top": 0,
          },
          Object {
            "height": 1334,
            "width": 600,
          },
        ]
      }
    >
      <View
        dataSet={Object {}}
        style={
          Array [
            Object {
              "backgroundColor": "#eeeeee",
              "flex": 1,
              "justifyContent": "space-between",
              "paddingBottom": 80,
              "paddingTop": 80,
            },
            Array [
              Object {},
            ],
          ]
        }
      >
        <View
          dataSet={Object {}}
          style={
            Array [
              Object {
                "borderBottomWidth": 1,
                "borderColor": "#7f7f7f",
                "paddingBottom": 8,
                "paddingLeft": 16,
                "paddingRight": 20,
                "paddingTop": 8,
              },
              Array [
                Object {},
              ],
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              dataSet={Object {}}
              style={
                Object {
                  "alignItems": "center",
                }
              }
            >
              <View
                dataSet={Object {}}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#a1a1aa",
                    "borderColor": "#27272a",
                    "borderRadius": 9999,
                    "height": 96,
                    "justifyContent": "center",
                    "position": "relative",
                    "width": 96,
                  }
                }
              >
                <Text
                  allowFontScaling={false}
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": undefined,
                        "fontSize": 40,
                      },
                      undefined,
                      Object {
                        "fontFamily": "FontAwesome",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
              </View>
              <View
                dataSet={Object {}}
                style={Object {}}
              />
              <Text
                dataSet={Object {}}
                style={
                  Object {
                    "backgroundColor": undefined,
                    "color": "#171717",
                    "fontFamily": undefined,
                    "fontSize": 18,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                    "lineHeight": 27,
                    "marginBottom": 8,
                    "marginTop": 8,
                    "textDecorationLine": undefined,
                  }
                }
              >
                undefined undefined
              </Text>
            </View>
          </View>
        </View>
        <RCTScrollView
          contentContainerStyle={
            Array [
              Object {},
              Object {
                "dataSet": Object {},
              },
            ]
          }
          data={
            Array [
              Object {
                "icon": <Icon
                  allowFontScaling={false}
                  color="#000000"
                  name="power-off"
                  size={24}
                />,
                "id": "LOGOUT",
                "label": "LOGOUT",
                "onPress": [Function],
              },
            ]
          }
          dataSet={Object {}}
          getItem={[Function]}
          getItemCount={[Function]}
          keyExtractor={[Function]}
          onContentSizeChange={[Function]}
          onLayout={[Function]}
          onMomentumScrollBegin={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEventThrottle={50}
          stickyHeaderIndices={Array []}
          style={Object {}}
          viewabilityConfigCallbackPairs={Array []}
        >
          <View>
            <View
              onLayout={[Function]}
              style={null}
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  dataSet={Object {}}
                  style={
                    Array [
                      Object {
                        "borderBottomWidth": 1,
                        "borderColor": "#7f7f7f",
                        "paddingBottom": 8,
                        "paddingLeft": 16,
                        "paddingRight": 20,
                        "paddingTop": 8,
                      },
                      Array [
                        Object {},
                      ],
                    ]
                  }
                >
                  <View
                    dataSet={Object {}}
                    style={
                      Object {
                        "flexDirection": "row",
                      }
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      selectable={false}
                      style={
                        Array [
                          Object {
                            "color": "#000000",
                            "fontSize": 24,
                          },
                          undefined,
                          Object {
                            "fontFamily": "FontAwesome",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      
                    </Text>
                    <View
                      dataSet={Object {}}
                      style={
                        Object {
                          "width": 12,
                        }
                      }
                    />
                    <View
                      dataSet={Object {}}
                      style={
                        Object {
                          "justifyContent": "center",
                        }
                      }
                    >
                      <Text
                        dataSet={Object {}}
                        style={
                          Object {
                            "backgroundColor": undefined,
                            "color": "#171717",
                            "fontFamily": undefined,
                            "fontSize": 14,
                            "fontStyle": "normal",
                            "fontWeight": "700",
                            "letterSpacing": 0,
                            "lineHeight": 21,
                            "textDecorationLine": undefined,
                          }
                        }
                      >
                        LOGOUT
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
      <View
        pointerEvents="none"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
</RNCSafeAreaProvider>
`;

exports[`EmailVerification should match success snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Array [
          Object {
            "borderWidth": 0,
            "left": 0,
            "position": "absolute",
            "top": 0,
          },
          Object {
            "height": 1334,
            "width": 750,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "position": "absolute",
            },
          ]
        }
      >
        <RNSVGSvgView
          bbHeight={1334}
          bbWidth={750}
          focusable={false}
          height={1334}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "flex": 0,
                "height": 1334,
                "width": 750,
              },
            ]
          }
          width={750}
        >
          <RNSVGGroup>
            <RNSVGPath
              d="M0 200 L0 300 L750 200 L750 100 L0 200 Z"
              fill={4289527808}
              propList={
                Array [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M0 350 L0 370 L750 270 L750 250 L0 350 Z"
              fill={4289527808}
              propList={
                Array [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M0 1084 L0 1104 L750 1004 L750 984 L0 1084 Z"
              fill={4278199178}
              propList={
                Array [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M0 1154 L0 1254 L750 1154 L750 1054 L0 1154 Z"
              fill={4278199178}
              propList={
                Array [
                  "fill",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
      <View
        dataSet={Object {}}
        style={
          Object {
            "display": "flex",
            "flex": 1,
            "flexDirection": "column",
          }
        }
      >
        <View
          dataSet={Object {}}
          style={
            Object {
              "display": "flex",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingBottom": 20,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 20,
            }
          }
        >
          <View
            dataSet={Object {}}
            elevation={1}
            shadowColor="#000000"
            shadowOffset={
              Object {
                "height": 1,
                "width": 0,
              }
            }
            shadowOpacity={0.8}
            shadowRadius={1}
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "#eeeeee",
                  "borderRadius": 20,
                  "flex": 1,
                  "paddingBottom": 20,
                  "paddingLeft": 20,
                  "paddingRight": 20,
                  "paddingTop": 20,
                },
                Array [
                  Object {},
                ],
              ]
            }
            testID="email-verification"
          >
            <View
              dataSet={Object {}}
              style={
                Array [
                  Object {
                    "display": "flex",
                    "flexDirection": "row",
                    "marginBottom": 20,
                  },
                  Array [
                    Object {},
                  ],
                ]
              }
              styleProps={
                Object {
                  "mb": 5,
                }
              }
            >
              <View
                dataSet={Object {}}
                style={
                  Array [
                    Object {
                      "backgroundColor": "#00238a",
                      "color": "#00238a",
                      "marginRight": 8,
                      "paddingRight": 8,
                    },
                    Array [
                      Object {},
                    ],
                  ]
                }
              />
              <Text
                dataSet={Object {}}
                style={
                  Array [
                    Object {
                      "backgroundColor": undefined,
                      "color": "#171717",
                      "fontFamily": undefined,
                      "fontSize": 20,
                      "fontStyle": "normal",
                      "fontWeight": "700",
                      "letterSpacing": 0,
                      "lineHeight": 30,
                      "textDecorationLine": undefined,
                    },
                    Array [
                      Object {},
                    ],
                  ]
                }
              >
                SUCCESS
                !
              </Text>
            </View>
            <View
              dataSet={Object {}}
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "justifyContent": "center",
                }
              }
            >
              <Image
                accessibilityLabel="newor"
                alt="newor"
                dataSet={Object {}}
                onError={[Function]}
                source={
                  Object {
                    "testUri": "../../../src/asset/image/newor-success.png",
                  }
                }
                style={
                  Object {
                    "height": 250,
                    "maxWidth": "100%",
                    "width": 250,
                  }
                }
              />
            </View>
            <Text
              dataSet={Object {}}
              style={
                Object {
                  "backgroundColor": undefined,
                  "color": "#171717",
                  "fontFamily": undefined,
                  "fontSize": 14,
                  "fontStyle": "normal",
                  "fontWeight": "400",
                  "letterSpacing": 0,
                  "lineHeight": 21,
                  "marginBottom": 20,
                  "textDecorationLine": undefined,
                }
              }
            >
              EMAIL_VERIFICATION_SUCCESS
            </Text>
            <View
              accessibilityRole="button"
              accessible={true}
              collapsable={false}
              colorScheme="primary"
              dataSet={Object {}}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ad0000",
                  "borderRadius": 4,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginBottom": 20,
                  "paddingBottom": 10,
                  "paddingLeft": 12,
                  "paddingRight": 12,
                  "paddingTop": 10,
                }
              }
              testID="login-submit"
            >
              <View
                dataSet={Object {}}
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  }
                }
                test={true}
              >
                <View
                  dataSet={Object {}}
                  style={Object {}}
                >
                  <Text
                    dataSet={Object {}}
                    style={
                      Object {
                        "backgroundColor": undefined,
                        "color": "#fafafa",
                        "fontFamily": undefined,
                        "fontSize": 14,
                        "fontStyle": "normal",
                        "fontWeight": "700",
                        "letterSpacing": 0,
                        "lineHeight": 21,
                        "textDecorationLine": undefined,
                      }
                    }
                  >
                    LOGIN
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        pointerEvents="none"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            undefined,
          ]
        }
      />
    </View>
    <View
      elevation={0}
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Array [
          Object {
            "borderWidth": 0,
            "left": -600,
            "position": "absolute",
            "top": 0,
          },
          Object {
            "height": 1334,
            "width": 600,
          },
        ]
      }
    >
      <View
        dataSet={Object {}}
        style={
          Array [
            Object {
              "backgroundColor": "#eeeeee",
              "flex": 1,
              "justifyContent": "space-between",
              "paddingBottom": 80,
              "paddingTop": 80,
            },
            Array [
              Object {},
            ],
          ]
        }
      >
        <View
          dataSet={Object {}}
          style={
            Array [
              Object {
                "borderBottomWidth": 1,
                "borderColor": "#7f7f7f",
                "paddingBottom": 8,
                "paddingLeft": 16,
                "paddingRight": 20,
                "paddingTop": 8,
              },
              Array [
                Object {},
              ],
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              dataSet={Object {}}
              style={
                Object {
                  "alignItems": "center",
                }
              }
            >
              <View
                dataSet={Object {}}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#a1a1aa",
                    "borderColor": "#27272a",
                    "borderRadius": 9999,
                    "height": 96,
                    "justifyContent": "center",
                    "position": "relative",
                    "width": 96,
                  }
                }
              >
                <Text
                  allowFontScaling={false}
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": undefined,
                        "fontSize": 40,
                      },
                      undefined,
                      Object {
                        "fontFamily": "FontAwesome",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
              </View>
              <View
                dataSet={Object {}}
                style={Object {}}
              />
              <Text
                dataSet={Object {}}
                style={
                  Object {
                    "backgroundColor": undefined,
                    "color": "#171717",
                    "fontFamily": undefined,
                    "fontSize": 18,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                    "lineHeight": 27,
                    "marginBottom": 8,
                    "marginTop": 8,
                    "textDecorationLine": undefined,
                  }
                }
              >
                undefined undefined
              </Text>
            </View>
          </View>
        </View>
        <RCTScrollView
          contentContainerStyle={
            Array [
              Object {},
              Object {
                "dataSet": Object {},
              },
            ]
          }
          data={
            Array [
              Object {
                "icon": <Icon
                  allowFontScaling={false}
                  color="#000000"
                  name="power-off"
                  size={24}
                />,
                "id": "LOGOUT",
                "label": "LOGOUT",
                "onPress": [Function],
              },
            ]
          }
          dataSet={Object {}}
          getItem={[Function]}
          getItemCount={[Function]}
          keyExtractor={[Function]}
          onContentSizeChange={[Function]}
          onLayout={[Function]}
          onMomentumScrollBegin={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEventThrottle={50}
          stickyHeaderIndices={Array []}
          style={Object {}}
          viewabilityConfigCallbackPairs={Array []}
        >
          <View>
            <View
              onLayout={[Function]}
              style={null}
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  dataSet={Object {}}
                  style={
                    Array [
                      Object {
                        "borderBottomWidth": 1,
                        "borderColor": "#7f7f7f",
                        "paddingBottom": 8,
                        "paddingLeft": 16,
                        "paddingRight": 20,
                        "paddingTop": 8,
                      },
                      Array [
                        Object {},
                      ],
                    ]
                  }
                >
                  <View
                    dataSet={Object {}}
                    style={
                      Object {
                        "flexDirection": "row",
                      }
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      selectable={false}
                      style={
                        Array [
                          Object {
                            "color": "#000000",
                            "fontSize": 24,
                          },
                          undefined,
                          Object {
                            "fontFamily": "FontAwesome",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      
                    </Text>
                    <View
                      dataSet={Object {}}
                      style={
                        Object {
                          "width": 12,
                        }
                      }
                    />
                    <View
                      dataSet={Object {}}
                      style={
                        Object {
                          "justifyContent": "center",
                        }
                      }
                    >
                      <Text
                        dataSet={Object {}}
                        style={
                          Object {
                            "backgroundColor": undefined,
                            "color": "#171717",
                            "fontFamily": undefined,
                            "fontSize": 14,
                            "fontStyle": "normal",
                            "fontWeight": "700",
                            "letterSpacing": 0,
                            "lineHeight": 21,
                            "textDecorationLine": undefined,
                          }
                        }
                      >
                        LOGOUT
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
      <View
        pointerEvents="none"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
</RNCSafeAreaProvider>
`;
